paddle/gserver/tests/test_Proposal.cpp

```diff
@@ -0,0 +1,163 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <gtest/gtest.h>
+#include <string>
+#include <vector>
+
+#include "LayerGradUtil.h"
+#include "paddle/testing/TestUtil.h"
+
+using namespace paddle;  // NOLINT
+using namespace std;     // NOLINT
+
+// Do one forward pass of priorBox layer and check to see if its output
+// matches the given result
+void doOneProposalTest(MatrixPtr& inputLoc,
+                       MatrixPtr& inputConf,
+                       MatrixPtr& inputAnchor,
+                       size_t feature_map_width,
+                       size_t feature_map_height,
+                       real nms_threshold,
+                       bool use_gpu,
+                       MatrixPtr& result) {
+  // Setting up the detection output layer
+  TestConfig configt;
+  configt.layerConfig.set_type("proposal");
+  LayerInputConfig* input = configt.layerConfig.add_inputs();
+  configt.layerConfig.add_inputs();
+  configt.layerConfig.add_inputs();
+
+  ProposalConfig* proposalConf = input->mutable_proposal_conf();
+  proposalConf->set_width(feature_map_width);
+  proposalConf->set_height(feature_map_height);
+  proposalConf->set_nms_threshold(nms_threshold);
+  proposalConf->set_confidence_threshold(0.01);
+  proposalConf->set_nms_top_k(20);
+  proposalConf->set_keep_top_k(10);
+  proposalConf->set_min_width(0);
+  proposalConf->set_min_height(0);
+  configt.inputDefs.push_back({INPUT_DATA_TARGET, "anchors", 28, 0});
+  configt.inputDefs.push_back({INPUT_DATA, "input_loc", 16, 0});
+  configt.inputDefs.push_back({INPUT_DATA, "input_conf", 8, 0});
+
+  // data layer initialize
+  std::vector<DataLayerPtr> dataLayers;
+  LayerMap layerMap;
+  vector<Argument> datas;
+  initDataLayer(
+      configt, &dataLayers, &datas, &layerMap, "anchors", 1, false, use_gpu);
+
+  dataLayers[0]->getOutputValue()->copyFrom(*inputAnchor);
```

Comment:
It seems that the size of dataLayers[*]->outputvalue is wrong.
You can use `LOG(INFO)<< "dataLayers[0] size: "<<dataLayers[0]->getOutputValue()->getElementCnt();` to check if the size  of OuptValue equals to size of inputAnchor.